### ISSUE TITLE
PIE794 should only remove simple code?

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#1798]
```
class Foo(object):
   bar = baz()
   bar = foo(bar)
```

`bar` uses `bar`. The fixer removes that to (or something else since #1794):
```
class Foo(object):
   bar = baz()
```

Even if the bar reference is not used I would say the code should not be changed as it makes function calls which might be necessary.

**Comment by charliermarsh** [comment]
Hmm. It's hard to think of logic that would work here, yet would still make the autofix behavior even marginally useful. The primary use-case is redefined fields in ORMs, which necessitate function calls. I guess in theory we could try to omit any call that relies on a variable bound within the same scope, but that will take a bit of time.

**Comment by charliermarsh** [comment]
Do you have any real-world examples of where this is triggering?

**Comment by charliermarsh** [comment]
Actually, here's what I would propose (which would also resolve the other issue you filed around this rule): we only autofix if the _values_ are the same. (Right now, we use that same logic for duplicate dictionary keys, so it feels very consistent.)
